.github/workflows/ci_cd.yml

```
@@ -822,13 +822,32 @@ jobs:
       - name: Run Playwright tests
         run: just frontend/run ${{ matrix.script }} --workers=2
 
+      - name: Check for changes after running tests with -u
+        id: check_snapshots_changed
+        run: |
+          if [ -n "$(git diff --binary --name-only)" ]; then
+            git add . && git diff --staged --binary | gzip -f > /tmp/snapshot_diff.patch.gz
+            echo "snapshots_changed=true" | tee "$GITHUB_OUTPUT"
+            git diff --binary --name-only
+          else
+            echo "snapshots_changed=false" | tee "$GITHUB_OUTPUT"
+          fi
+
       - uses: actions/upload-artifact@v4
         if: failure()
         id: test-results
         with:
           name: ${{ matrix.name }}_test_results
           path: frontend/test-results
 
+      # This step only runs if there was a failure and snapshots indeed changed.
+      - uses: actions/upload-artifact@v4
+        if: failure() && steps.check_snapshots_changed.outputs.snapshots_changed == 'true'
```

When passing `-u`, the exit code will be 0 (if there are no regular test failures), so in the case of when there are _only_ updated snapshots, checking `failure()` would only upload the snapshots if there were additional failing tests.

```suggestion
        if: steps.check_snapshots_changed.outputs.snapshots_changed == 'true'
```